src/test/java/net/imagej/ops/image/equation/CoordinateEquationTest.java

```diff
@@ -0,0 +1,176 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.ops.image.equation;
+
+import static org.junit.Assert.assertEquals;
+
+import net.imagej.ops.AbstractOpTest;
+import net.imagej.ops.special.function.AbstractUnaryFunctionOp;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imglib2.Dimensions;
+import net.imglib2.FinalDimensions;
+import net.imglib2.IterableInterval;
+import net.imglib2.RandomAccess;
+import net.imglib2.img.Img;
+import net.imglib2.type.numeric.integer.IntType;
+import net.imglib2.type.numeric.integer.ShortType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.type.numeric.real.FloatType;
+
+import org.junit.Test;
+
+/**
+ * Test XY, and coordinate based equations that is equations of the form f(x,y)
+ * or f(c[0], [1], ... c[n]) where c represents pixel coordinates
+ * 
+ * @author bnorthan
+ */
+public class CoordinateEquationTest extends AbstractOpTest {
+
+	long[] size = new long[] { 100, 100 };
+
+	// define start and end
+	double[] start = new double[] { -1., -1. };
+	double[] end = new double[] { 1., 1. };
+
+	// calculate spacing based on size and starting point
+	double[] spacing = new double[] { (end[0] - start[0]) / (size[0] - 1),
+		(end[1] - start[1]) / (size[1] - 1) };
+
+	Dimensions dimensions = new FinalDimensions(size[0], size[1]);
+
+	/**
+	 * Test the lambda version of the XY equation
+	 */
+	@Test
+	public void testEquationXY() {
+
+		// create a double image
+		final IterableInterval<DoubleType> image1 = ops.create().img(dimensions,
+			new DoubleType());
+
+		ops.image().equation(image1, (x, y) -> 10 * (Math.cos(0.3 * x) + Math.sin(
+			0.3 * y)));
+
+		// assert against value determined by running string version of equation in
+		// jython
+		assertEquals(ops.stats().sum(image1).getRealDouble(), 446.1755977585166,
+			0.001);
+
+		// create an integer type image
+		final IterableInterval<IntType> image2 = ops.create().img(dimensions,
+			new IntType());
+
+		// implement x^2+y^2
+		ops.image().equation(image2, (x, y) -> Math.pow(x, 2) + Math.pow(y, 2));
+
+		assertEquals(ops.stats().sum(image2).getRealDouble(), 6.567e7, 0.001);
+
+		// create a float image
+		final IterableInterval<FloatType> image3 = ops.create().img(dimensions,
+			new FloatType());
+
+		// implement x^2+y^2 taking into account the calibration (start and spacing)
+		ops.image().equation(image3, (x, y) -> Math.pow(start[0] + spacing[0] * x,
+			2) + Math.pow(start[1] + spacing[1] * y, 2));
```

Comment:
Here you should use `ops().run(net.imagej.ops.image.equation.DefaultXYEquation.class, out, in)`, to ensure that `DefaultXYEquation` is always being tested and not some other op.

This isn't an issue now, but it could become one later if someone creates another `CoordinatesEquationOp`.